frontend/src/metabase/common/components/Sidesheet/Sidesheet.tsx

```diff
@@ -0,0 +1,79 @@
+import type React from "react";
+import { t } from "ttag";
+
+import { Modal, Stack } from "metabase/ui";
+
+import Styles from "./sidesheet.module.css";
+
+type Size = "xs" | "sm" | "md" | "lg" | "xl" | "auto";
+
+interface SidesheetProps {
+  title?: React.ReactNode;
+  isOpen: boolean;
+  onClose: () => void;
+  size?: Size;
+  children: React.ReactNode;
+  /** use this if you want to enable interior scrolling of tab panels */
+  removeBodyPadding?: boolean;
+}
+
+const sizes: Record<Size, string> = {
+  xs: "20rem",
+  sm: "30rem",
+  md: "40rem",
+  lg: "50rem",
+  xl: "60rem",
+  auto: "auto",
+};
+
+export function Sidesheet({
+  title,
+  isOpen,
+  onClose,
+  size = "sm",
+  children,
+  removeBodyPadding,
+}: SidesheetProps) {
+  return (
+    <Modal.Root opened={!!isOpen} onClose={onClose} h="100dvh">
+      <Modal.Overlay aria-label={t`modal overlay`} />
```

Comment:
Not sure overlays need aria labels - a screen reader would never announce them

Maybe a testid could work here. It's one less string in poeditor